Makefile

```diff
@@ -132,18 +133,32 @@ cli-install:
 clean:  cli-clean test-clean manager-clean deploy-clean
 
 .PHONY: docker-build
-# Build the docker image
+# Build the docker image for each supported platform
 docker-build: generate lint
-	docker build --build-arg ldflags_arg="${LDFLAGS}" . -t ${DOCKER_IMG}:${DOCKER_TAG}
-	docker tag ${DOCKER_IMG}:${DOCKER_TAG} ${DOCKER_IMG}:v${GIT_VERSION}
-	docker tag ${DOCKER_IMG}:${DOCKER_TAG} ${DOCKER_IMG}:latest
+	$(foreach arch,$(SUPPORTED_PLATFORMS),docker build --build-arg ldflags_arg="$(LDFLAGS)" -f Dockerfile.$(arch) -t $(DOCKER_IMG)-$(arch):$(DOCKER_TAG) .;)
+	$(foreach arch,$(SUPPORTED_PLATFORMS),docker tag $(DOCKER_IMG)-$(arch):$(DOCKER_TAG) $(DOCKER_IMG)-$(arch):v$(GIT_VERSION);)
+	$(foreach arch,$(SUPPORTED_PLATFORMS),docker tag $(DOCKER_IMG)-$(arch):$(DOCKER_TAG) $(DOCKER_IMG)-$(arch):latest;)
+
+.PHONY: docker-push-all-platforms
+# Push the platform specific images for each supported arch
+docker-push-platforms:
+	$(foreach arch,$(SUPPORTED_PLATFORMS),docker push $(DOCKER_IMG)-$(arch):$(DOCKER_TAG);)
+	$(foreach arch,$(SUPPORTED_PLATFORMS),docker push $(DOCKER_IMG)-$(arch):v$(GIT_VERSION);)
+	$(foreach arch,$(SUPPORTED_PLATFORMS),docker push $(DOCKER_IMG)-$(arch):latest;)
+
+.PHONY: docker-push-manifest
+# Push the multi-arch image manifest
+docker-push-manifests:
```

Comment:
Why use experimental features here? Do we need image manifests here? I'm curious, because I haven't use this before.

Reply:
This is a simple implementation and others in the Kubernetes community utilize it for now while the support for multi-platform manifests generally matures. You can see similar examples in Velero: https://github.com/vmware-tanzu/velero/blob/master/Makefile#L186, so we wont be pioneers in this. I have also used `docker buildx` options but I find these to be a bit less simple to use and reason about at this time than simply directly creating the manifests.

From the end user perspective manifests are essentially just architecture-based links to other images such that every arch can `docker run -it kudobuilder/kudo` and under the hood it's a pointer to the `-${ARCH}` image (which with this PR we will start pushing as well), so even though technically the CLI functionality is "experimental" in recent versions of `docker`, any CI tests that deploy the controller container with the manifest image (e.g. `kudobuilder/kudo`) will be a valid test to make sure this is working properly.

Reply:
Interesting, thank for the explanation!